docs/blog/20240124-mauricioszabo-the-quest-for-electron-lts.md

```diff
@@ -0,0 +1,86 @@
+
+Fortunately, we found a fix - there was an experimental, and incomplete, version of Superstring that compiled to Web Assembly - that means, a new format that _every browser_ should run without any problem, and that's how we ported Superstring to run newer Electron.... for a while.
+
+## More things break
+
+Our syntax highlighter tree-sitter also didn't work on newer Electron version - on Electron 14, it stopped working completely. So, for the time being, we disabled it, and kept trying to port things, fix stuff, etc. Fortunately, all other libraries were ready to be used in newer Electron contexts, and we just had to fix some "synchronous" calls to be "asynchronous" and that worked.
+
+I wasn't that worried about disabling tree-sitter because, at the time, we were already trying to make a modern tree-sitter implementation, so I knew that sometime in the future, we could re-enable tree-sitter and that it would work (spoiler alert: **it DID!**).
+
+Then I got greedy, and decided to make Pulsar work in the newer Electron at the time - Electron 23.
+
+## Things work fine! (juuuust kidding!)
+
+Well, if you have been following the tone of this post... of course you know what happens next, right?
+
+Electron 21 enabled something called _V8 memory cage_. I honestly don't fully understand that that means, but what ended up happening is that _another native library_ was crashing the editor - and the name of the library is Oniguruma.
+
+If you're not aware of what this is, it's a regular expression library. If you _also don't know_ what that is, basically is the library used in TextMate grammars to do syntax coloring.
+
+So, summarizing: Atom (and by definition, Pulsar) had two ways of highlighting code (coloring the keywords, functions, classes, methods, etc): one is called tree-sitter, that breaks in Electron 14, and another callled TextMate, that breaks in Electron 21...
+
+...great.
+
+A code editor without syntax highlighting is probably not a good idea at all; we could not disable it and basically offer no syntax highlighting library. So, I started a quest to migrate TextMate grammars to something different - and which better place to start than VSCode?
+
+It turns out that VSCode uses a WASM version of Oniguruma - that they called [vscode-oniguruma](https://github.com/microsoft/vscode-oniguruma) - and it's MIT-licensed too, which is compatible with the Pulsar's license. And luckily for us, the Pulsar's highlighter (again, inherited from Atom) is on its own separate library, well-tested, called [first-mate](https://github.com/atom/first-mate). So, I did a wordplay and called the new library `second-mate` (and up to this day, I still reget not calling it `check-mate` to be honest) that uses this new WASM version. After fixing some infinite loops and other edge-cases that were not fully tested on the original project, I plugged `second-mate` in the editor, and as unreal that this might sound, it worked!
```

Comment:
```suggestion
It turns out that VSCode uses a WASM version of Oniguruma - that they called [vscode-oniguruma](https://github.com/microsoft/vscode-oniguruma) - and it's MIT-licensed too, which is compatible with Pulsar's license. And luckily for us, Pulsar's highlighter (again, inherited from Atom) is on its own separate library, well-tested, called [first-mate](https://github.com/atom/first-mate). So, I did a wordplay and called the new library `second-mate` (and up to this day, I still reget not calling it `check-mate` to be honest) that uses this new WASM version. After fixing some infinite loops and other edge-cases that were not fully tested on the original project, I plugged `second-mate` in the editor, and as unreal that this might sound, it worked!
```